### ISSUE TITLE
Outdated documentation for type conversions

### ISSUE THREAD

**Issue by Vectornaut** [JuliaLang/julia#53243]
### Description

The instructions for [defining new conversions](https://docs.julialang.org/en/v1/manual/conversion-and-promotion/#Defining-New-Conversions) no longer work as written, because you can no longer expand the definition of a base method without qualifying its name.

### Reproduction

1. Start the REPL.
2. In the REPL, define a type.
```julia
struct MyType
  value
end
```
3. Read the instructions for [defining new conversions](https://docs.julialang.org/en/v1/manual/conversion-and-promotion/#Defining-New-Conversions).
4. In the REPL, follow the instructions to define a new type conversion.
```julia
convert(::Type{MyType}, x) = MyType(x)
```

### Expected behavior

A new conversion is defined.

### Actual behavior

```
ERROR: invalid method definition in Main: function Base.convert must be explicitly imported to be extended
```

### Additional context

The problem is even worse when the new conversion is defined within a module, because then you don't get the "Invalid method definition" error. Instead, the new conversion simply fails to extend the definition of `Base.convert`, making it much less clear why it can't be found when you try to use it. I'm not exactly new to Julia, and I just spent twenty minutes debugging a problem that turned out to be caused by this.

At the very least, the example of defining a new conversion should be something that works as written, like these:
```julia
Base.convert(::Type{MyType}, x) = MyType(x)
```
```julia
import Base.convert

convert(::Type{MyType}, x) = MyType(x)
```

**Comment by inkydragon** [comment]
Looks like it's working fine on v1.10.0.

```julia
julia> struct MyType
         value
       end

julia> convert(::Type{MyType}, x) = MyType(x)
convert (generic function with 1 method)

julia> convert(MyType, 1)
MyType(1)

julia> convert(MyType, pi)
MyType(π)

julia> versioninfo()
Julia Version 1.10.0
Commit 3120989f39 (2023-12-25 18:01 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 32 × 13th Gen Intel(R) Core(TM) i9-13900HX
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, goldmont)
  Threads: 1 on 32 virtual cores
```

And latest doc already imports `Base.convert` at the beginning.
Maybe the doc update needs to be backported. (#50274)

https://github.com/JuliaLang/julia/blob/a6ce761fd3c345c3c17d52d0a972bbc1cb325197/doc/src/manual/conversion-and-promotion.md?plain=1#L167-L170

**Comment by Vectornaut** [comment]
Whoops—I left out a step when I wrote the reproduction procedure, and never tried it again from a clean REPL! I've updated the procedure.

It turns out that before doing the original reproduction procedure, I tried to do an impossible conversion, like one of these:
```julia
convert(Int64, "a")
```
```julia
convert(Int64, 1.5)
```
This step is necessary to get the "invalid method definition" error later. Maybe this is expected behavior, although I certainly didn't expect it!

**Comment by jishnub** [comment]
Yes, once you use the Base convert function, you can't add a new function by the same name (as in the example above). You can only add methods to it. And to add methods to the function, you either need to import it, or fully qualify the name.

I think it's a good idea for the docs to be explicit about this, and add methods to `Base.convert`.
